src/ray/raylet/node_manager.cc

```diff
@@ -957,6 +1017,38 @@ void NodeManager::HandleObjectMissing(const ObjectID &object_id) {
   }
 }
 
+void NodeManager::ForwardTaskOrResubmit(const Task &task,
+                                        const ClientID &node_manager_id) {
+  /// TODO(rkn): Should we check that the node manager is remote and not local?
+  /// TODO(rkn): Should we check if the remote node manager is known to be dead?
+  const TaskID task_id = task.GetTaskSpecification().TaskId();
+
+  // Attempt to forward the task.
+  if (!ForwardTask(task, node_manager_id).ok()) {
+    RAY_LOG(INFO) << "Failed to forward task " << task_id << " to node manager "
+                  << node_manager_id;
+
+    // Create a timer to resubmit the task in a little bit.
+    boost::asio::deadline_timer timer(io_service_);
```

Comment:
@stephanie-wang `io_service_` is the correct service (in fact, the only one, right?) to use for retrying the task submission, right?

Reply:
Yup!